addons/website_sale/tests/test_delivery_ui.py

```diff
@@ -72,3 +79,25 @@ def test_01_free_delivery_when_exceed_threshold(self):
         }])
 
         self.start_tour("/", 'check_free_delivery', login="admin")
+
+    def test_pay_button_disabled_when_carrier_has_error(self):
+        if self.env['ir.module.module']._get('payment_custom').state != 'installed':
+            self.skipTest("Transfer provider is not installed")
+        Monetary = self.env['ir.qweb.field.monetary']
+        usd_currency = self.env.ref('base.USD')
+        with patch.object(WebsiteSaleDelivery, '_get_rate',
+                          lambda controller, *args, **kwags: {
```

Comment:
kwargs*

Actually, if you don't need the params in your mock, you could just use the `return_value` param to keep it clear.

Reply:
using `return_value` on controller functions causes some runbot wanings.

Reply:
https://runbot.odoo.com/runbot/build/61758589 

according to my teammates, the issue is that if you don't give a second parameter (or third parameter for patch.object) the method is replaced by Mock object instance (a magic object that has all attributes existing), then when we generates routes, we will find the overridden method (the Mock object) to be a route when it was not a route in reality because of how Mock objects are.